common/src/main/java/com/microsoft/identity/common/internal/result/AdalBrokerResultAdapter.java

```diff
@@ -0,0 +1,343 @@
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+package com.microsoft.identity.common.internal.result;
+
+import android.accounts.AccountManager;
+import android.net.Uri;
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.text.TextUtils;
+
+import com.microsoft.identity.common.adal.internal.ADALError;
+import com.microsoft.identity.common.adal.internal.AuthenticationConstants;
+import com.microsoft.identity.common.exception.ArgumentException;
+import com.microsoft.identity.common.exception.BaseException;
+import com.microsoft.identity.common.exception.ClientException;
+import com.microsoft.identity.common.exception.ErrorStrings;
+import com.microsoft.identity.common.exception.IntuneAppProtectionPolicyRequiredException;
+import com.microsoft.identity.common.exception.ServiceException;
+import com.microsoft.identity.common.exception.UserCancelException;
+import com.microsoft.identity.common.internal.cache.SchemaUtil;
+import com.microsoft.identity.common.internal.dto.IAccountRecord;
+import com.microsoft.identity.common.internal.logging.Logger;
+import com.microsoft.identity.common.internal.util.HeaderSerializationUtil;
+
+import static com.microsoft.identity.common.adal.internal.AuthenticationConstants.Broker.CliTelemInfo.RT_AGE;
+import static com.microsoft.identity.common.adal.internal.AuthenticationConstants.Broker.CliTelemInfo.SERVER_ERROR;
+import static com.microsoft.identity.common.adal.internal.AuthenticationConstants.Broker.CliTelemInfo.SERVER_SUBERROR;
+import static com.microsoft.identity.common.adal.internal.AuthenticationConstants.Broker.CliTelemInfo.SPE_RING;
+
+public class AdalBrokerResultAdapter implements IBrokerResultAdapter {
+
+    private static final String TAG = AdalBrokerResultAdapter.class.getName();
+
+    @Override
+    public Bundle bundleFromAuthenticationResult(@NonNull final ILocalAuthenticationResult authenticationResult) {
+
+        Logger.verbose(TAG , "Constructing success bundle from Authentication Result.");
+
+        final Bundle resultBundle = new Bundle();
+
+        IAccountRecord accountRecord = authenticationResult.getAccountRecord();
+        resultBundle.putString(
+                AuthenticationConstants.Broker.ACCOUNT_LOGIN_HINT,
+                accountRecord.getUsername()
+        );
+        resultBundle.putString(
+                AuthenticationConstants.Broker.ACCOUNT_USERINFO_USERID,
+                accountRecord.getLocalAccountId()
+        );
+        resultBundle.putString(
+                AuthenticationConstants.Broker.ACCOUNT_USERINFO_USERID_DISPLAYABLE,
+                accountRecord.getUsername()
+        );
+        resultBundle.putString(
+                AuthenticationConstants.Broker.ACCOUNT_USERINFO_GIVEN_NAME,
+                accountRecord.getName()
```

Comment:
Should it be `accountRecord.getFirstName()`?

Reply:
It was always `getName()` before which corresponds to Adal's userinfo `getGivenName`  . So I think it's appropriate. Let me know what you think